ArcGISRuntimeSDKQt_QMLSamples/Features/FeatureLayer_Query/FeatureLayer_Query.qml

```diff
@@ -127,15 +96,18 @@ Rectangle {
                 selectByMouse: true
                 validator: RegExpValidator{ regExp: /^[a-zA-Z ]*$/ }
                 Keys.onReturnPressed: {
-                    query();
+                    btn.clicked();
                 }
             }
 
             Button {
+                id: btn
                 text: "Find and Select"
                 enabled: featureTable.loadStatus === Enums.LoadStatusLoaded
                 onClicked: {
-                    query();
+                    queryFeatures(findText.text, featureTable) // query the table
+                    .then(result=>selectResults(result, featureLayer)) // then select the features
```

Comment:
my idea was that the function would be a self contained function that took in a query string and a table and did what it needed - doesn't depend on things outside the scope of it. That way it can be used as a template for people to pull out into their own apps. I need to refactor something else to make that feasible still